credential.c

```diff
@@ -323,6 +332,20 @@ static int run_credential_helper(struct credential *c,
 
 	if (want_output) {
 		int r;
+		if (select_timeout.tv_sec) {
+			fd_set readfds;
+			FD_ZERO(&readfds);
+			FD_SET(helper.out, &readfds);
+
+			select(1, &readfds, NULL, NULL, &select_timeout);
+
+			if (!FD_ISSET(helper.out, &readfds)) {
+				/* Timeout complete before helper.out has bytes to read. */
+				kill_child_command(&helper);
+				return -1;
+			}
+		}
+
```

Comment:
I'm assuming that if we get a single byte from GCM, we'll get the full response so we don't need to do something more fancy.  GCM will either be stuck or not before the first byte is returned.

Reply:
That's also my expectation. The response is not provided until after the "blocking" part of GCM is complete. @mjcheetham or @ldennington might be able to confirm for sure.

Reply:
GCM only writes back to `stdout` once it has the complete result. If we've written anything back, we will soon exit.